### DISCUSSION TITLE
A tutorial on automatic differentiation with complex numbers

### DISCUSSION THREAD

**Discussion by pnkraemer** [jax-ml/jax#23898] [Show and tell]
Hi! :wave: 

I've recently written a tutorial on how to do forward- and reverse-mode automatic differentiation with complex numbers:

https://arxiv.org/abs/2409.06752

It might be interesting for everyone looking for an extended version of the "Complex numbers" section of JAX's autodiff cookbook (e.g., Discussion #11072?). The tutorial discusses linearisation, transposition, and Wirtinger calculus using only minimal complex analysis. It also contrasts different gradient conventions for complex numbers (Related: Issue #4891). 

All code examples use JAX. I hope it helps someone! :blush:

**Comment by pablo2909** [comment]
That's amazing thank you so much !
